gum/backend-libdwarf/gumsymbolutil-libdwarf.c

```diff
@@ -534,6 +601,32 @@ gum_get_function_addresses (void)
   return gum_function_addresses;
 }
 
+static GHashTable *
+gum_get_address_symbols (void)
+{
+  gboolean need_update;
+
+  gum_symbol_util_ensure_initialized ();
+
+  if (gum_cache_timer == NULL)
+  {
+    gum_cache_timer = g_timer_new ();
+
+    need_update = TRUE;
+  }
+  else
+  {
+    need_update = g_timer_elapsed (gum_cache_timer, NULL) >= GUM_MAX_CACHE_AGE;
+  }
+
+  if (need_update)
+  {
+    gum_process_enumerate_modules (gum_collect_module_functions, NULL);
+  }
```

Comment:
This common part should be extracted into a function and called from both. E.g. `gum_maybe_refresh_symbol_caches ()`